src/components/Achievement/Achievement.test.js

```diff
@@ -1,16 +1,47 @@
 import React from 'react'
-import { shallow } from 'enzyme'
+import { render, screen } from '@testing-library/react'
+
+import FarmhandContext from '../../Farmhand.context'
 
 import Achievement from './Achievement'
 
-let component
+describe('<Achievement />', () => {
+  let achievementObject
 
-beforeEach(() => {
-  component = shallow(
-    <Achievement {...{ achievement: {}, completedAchievements: {} }} />
-  )
-})
+  beforeEach(() => {
+    achievementObject = {
+      description: 'the best achievement',
+      id: 'achievement-1',
+      name: 'achievement one',
+      rewardDescription: 'the greatest reward',
+    }
+
+    const farmhandContextValue = {
+      gameState: {},
+      handlers: {},
+    }
+
+    render(
+      <FarmhandContext.Provider value={farmhandContextValue}>
+        <Achievement
+          achievement={achievementObject}
+          completedAchievements={{}}
+        />
+      </FarmhandContext.Provider>
+    )
+  })
+
+  test('renders the name of the achievement', () => {
+    expect(screen.getByText(achievementObject.name)).toBeInTheDocument()
+  })
+
+  test('renders the achievement description', () => {
+    expect(screen.getByText(achievementObject.description)).toBeInTheDocument()
+  })
 
-test('renders', () => {
-  expect(component).toHaveLength(1)
+  test('renders the reward description', () => {
+    expect(
+      screen.getByText(new RegExp(achievementObject.rewardDescription))
```

Comment:
How come a `RegExp` was needed here, but not for the other assertions?

Reply:
it's because there is more text inside the element (if i recall it's "Reward:\n${rewardDescription}") so the regex is required to tell it that it's ok it's not the entire text of the element (even though the description string isn't broken apart). i actually only found out because i started with the string matcher on this one

Reply:
Ah gotcha. This is something about React Testing Library I don't love. When I run into this I use `exact: false`:

```diff
diff --git a/src/components/Achievement/Achievement.test.js b/src/components/Achievement/Achievement.test.js
index 0d0877a..6853ad2 100644
--- a/src/components/Achievement/Achievement.test.js
+++ b/src/components/Achievement/Achievement.test.js
@@ -41,7 +41,7 @@ describe('<Achievement />', () => {
 
   test('renders the reward description', () => {
     expect(
-      screen.getByText(new RegExp(achievementObject.rewardDescription))
+      screen.getByText(achievementObject.rewardDescription, { exact: false })
     ).toBeInTheDocument()
   })
 })
```

I'm not sure if there's a benefit to one over the other, but that's worked out for me so far!

Reply:
oh, neat! i haven't discovered that but i like that more. will update next time i'm in the area